frontend/lib/src/features/transaction_history/transaction_history.dart

```diff
@@ -46,61 +45,80 @@ class TransactionHistory extends StatelessWidget {
             ),
             Padding(
               padding: const EdgeInsets.fromLTRB(25, 35, 25, 110),
-              child: Obx(
-                () => Column(
-                  children: [
-                    Dropdown(
-                      labelText: 'Filter transaction by type',
-                      items: TransactionTypes.values
-                          .map((type) => capitalizeFirstLetter(type.name))
-                          .toList(),
-                      selectedValue: capitalizeFirstLetter(
+              child: Column(
+                children: [
+                  Dropdown(
+                    labelText: 'Filter transaction by type',
+                    items: TransactionTypes.values
+                        .map((type) => capitalizeFirstLetter(type.name))
+                        .toList(),
+                    selectedValue: capitalizeFirstLetter(
+                      controller.selectedTransactionType.name,
+                    ),
+                    onChanged: (value) {
+                      controller.setSelectedTransactionType =
+                          TransactionTypes.values.firstWhere(
+                        (type) => capitalizeFirstLetter(type.name) == value,
+                      );
+                      debugPrint(
                         controller.selectedTransactionType.name,
-                      ),
+                      );
+                    },
+                  ),
+                  const SizedBox(height: 20),
+                  Obx(
```

Comment:
Please confirm, is it not possible to wrap as one all fields instead of individually?